test/e2e/common.go

```diff
@@ -328,6 +328,7 @@ func createApplyClusterTemplateInput(specName string, changes ...func(*clusterct
 		WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
 		WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 		WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+		WaitForMachinePools:          e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
```

Comment:
This is my bad for not adding to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/azure_test.go#L860 as part of https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/702bb03bc79b333ac6469d6b5649ee697d70bb30#diff-5486506a2ff6684dead6e32865402e057ed51f3af46f10cf8b14bfb1646830fbR6

This should have an equivalent effect since it sets a default for all calls to `createApplyClusterTemplateInput`, looks like the other templates using machine pools are currently each adding `withMachinePoolInterval`